en/contributing.md

```diff
@@ -28,171 +30,261 @@ and listed here:
-to test the docs properly you will need to have a web server set up. See
-[Ubuntu and Apache](https://help.ubuntu.com/lts/serverguide/httpd.html). The
-DocumentRoot should be the `htmldocs` directory:
+```no-highlight
+[img__webui_descriptor]: ../media/filename__webui_descriptor.png
+```
 
-```bash
-sudo cp -R htmldocs /var/www/htmldocs
+Explanation:
+
+- `filename`: name of file containing the image (omit the extension '.md')
+- `webui`: version of MAAS corresponding to the image of the web UI
+- `alt-attribute`: text that is shown in place of the image if the latter
+  cannot be displayed for some reason
+
+For example:
+
+```no-highlight
+![enable dhcp][img__2.1_enable-dhcp]
+![enable fire alarm][img__enable-fire-alarm]
+
+...
+
+[img__2.1_enable-dhcp]: ../media/installconfig-networking-dhcp__2.1_enable-dhcp.png
+[img__enable-fire-alarm]: ../media/installconfig-networking-dhcp__enable-fire-alarm.png
 ```
 
-You can then point your web browser at your local machine (127.0.0.1/htmldocs)
-to view the files.
+If the image is not of the MAAS web UI then simply omit the version part (e.g.
+do not include '2.1\_' anywhere in the above).
 
-Alternatively, you can use Python to start a simple HTTP server on the docs
-directory. Navigate to the `/htmldocs` directory of the docs and run the
-following:
+
+## Filenames
+
+The naming of files is based upon its location in the menu (see below). This
+makes it easier for the reader and the writer to build up a mental model of how
+everthing is structured.
+
+For example, the file `installconfig-networking-dhcp.md` is found under
+'Install & Configure' and then 'Networking'.
+
+
+## Navigation menu
+
+Adding a page (file) to the documentation may require the altering of
+`metadata.yaml`. Doing so will insert an entry into the left navigation pane
+(the menu) on the website.
+
+This is considered a major change so ensure your PR (pull request) includes a
+comment highlighting this change and why it is needed.
+
+
+## Build and view the HTML
+
+First install the builder. On Ubuntu 16.04 LTS:
 
 ```bash
-python -m SimpleHTTPServer
+sudo snap install documentation-builder
 ```
 
+To build the HTML, while in the root of the MAAS docs repository:
+
+```bash
+/snap/bin/documentation-builder
+```
 
-# Style and language
+See the [documentation-builder GitHub project][github-documentation-builder]
+for details.
 
-We are putting together a more comprehensive style guide, but for the moment the
-following are good guidelines:
+You will now need a web server. See the
+[Ubuntu Server Guide][ubuntu-serverguide-apache] for instructions on setting up
+Apache. The DocumentRoot should be the `build` directory. To test, point your
+browser at:
+
+```no-highlight
+http://127.0.0.1/en/contributing.html
+```
+
+Alternatively, you can use Python to start a simple HTTP server on the docs
+directory. While in the `build` directory run:
+
+```bash
+python -m SimpleHTTPServer
+```
```

Comment:
It may be worth mentioning this defaults to port 8000, and that to do the equivalent with Python 3 it's `python3 -m http.server`